pontoon/base/models.py

```diff
@@ -1401,6 +1391,37 @@ def can_commit(self):
         """True if we can commit strings back to this repo."""
         return self.type in ('svn', 'git', 'hg')
 
+    @property
+    def api_config(self):
+        """
+        Repository API configuration consists of:
+        - Endpoint: A URL pattern to get repository metadata from. Used during sync for faster
+          retrieval of latest commit hashes in combination with the Key.
+          Supports {locale_code} wildcard.
+        - Key: A string used to retrieve the latest commit hash from the JSON response.
+        """
+        url = self.url
+
+        if url.startswith('ssh://hg.mozilla.org/'):
+            endpoint_base = url.replace('ssh://', 'https://')
+            return {
+                'endpoint': os.path.join(endpoint_base, 'json-rev/default'),
+                'key': 'node',
```

Comment:
Alternative option: https://pypi.python.org/pypi/jsonpath/ . However, that may be overkill.